datafusion/functions/src/datetime/to_date.rs

```diff
@@ -47,22 +57,39 @@ impl ToDateFunc {
         match args.len() {
             1 => handle::<Date32Type, _, Date32Type>(
                 args,
+                // Although not being speficied, previous implementations have supported parsing date values that are followed
+                // by a time part separated either a whitespace ('2023-01-10 12:34:56.000') or a 'T' ('2023-01-10T12:34:56.000').
+                // Parsing these strings with "%Y-%m-%d" will fail. Therefore the string is split and the first part is used for parsing.
                 |s| {
-                    string_to_timestamp_nanos_shim(s)
-                        .map(|n| n / (1_000_000 * 24 * 60 * 60 * 1_000))
-                        .and_then(|v| {
-                            v.try_into().map_err(|_| {
-                                internal_datafusion_err!("Unable to cast to Date32 for converting from i64 to i32 failed")
-                            })
-                        })
+                    let string_to_parse;
+                    if s.find('T').is_some() {
+                        string_to_parse = s.split('T').next();
+                    } else {
+                        string_to_parse = s.split_whitespace().next();
+                    }
+                    if string_to_parse.is_none() {
+                        return Err(internal_datafusion_err!(
+                            "Cannot cast emtpy string to Date32"
+                        ));
+                    }
+
+                    // For most cases 'Date32Type::parse' would also work here, as it semantically parses
+                    // assuming a YYYY-MM-DD format. However 'Date32Type::parse' cannot handle negative
+                    // values for year (e.g. years in BC). Date32Type::parse_formatted can handle also these.
+                    match Date32Type::parse_formatted(string_to_parse.unwrap(), "%Y-%m-%d") {
+                    Some(v) => Ok(v),
```

Comment:
formatting seems off? typically `match` options are indented 
(`cargo fmt` seems to be fine either way, not sure why)